tdp/core/collection.py

```diff
@@ -36,67 +36,47 @@ class MissingMandatoryDirectoryError(Exception):
 class Collection:
     """An enriched version of an Ansible collection.
 
-    A TDP Collection is a directory containing playbooks, DAGs, default variables and schema variables.
-
-    Attributes:
-        path: Path to the collection.
-        name: Name of the collection.
-        dag_directory: Path to the DAG directory.
-        default_vars_directory: Path to the default variables directory.
-        operations_directory: Path to the operations directory.
-        schema_directory: Path to the schema variables directory.
-        dag_yamls: List of DAG files in the YAML format.
-        operations: Dictionary of operations with the operation name as key and the operation path as value.
+    A TDP Collection is a directory containing playbooks, DAGs, default variables and variables schemas.
     """
 
     def __init__(self, path: Union[str, os.PathLike]):
-        """Initialize a collection from a path.
+        """Initialize a collection.
 
         Args:
             path: The path to the collection.
+
+        Raises:
+            PathDoesNotExistsError: If the path does not exists.
+            PathIsNotADirectoryError: If the path is not a directory.
+            MissingMandatoryDirectoryError: If the collection does not contain a mandatory directory.
         """
         self._path = Path(path)
+        self._check_path()
+
         self._dag_yamls: Optional[List[Path]] = None
-        self._operations: Optional[Dict[str, Path]] = None
+        self._playbooks: Optional[Dict[str, Path]] = None
 
     @staticmethod
-    def from_path(path: Union[str, os.PathLike]):
-        """Factory method to build a collection from a path.
+    def from_path(path: Union[str, os.PathLike]) -> "Collection":
+        """Factory method to create a collection from a path.
 
         Args:
             path: The path to the collection.
 
-        Returns:
-            A collection.
-
-        Raises:
-            PathDoesNotExistsError: If the path does not exists.
-            PathIsNotADirectoryError: If the path is not a directory.
-            MissingMandatoryDirectoryError: If the path does not contain the mandatory directories.
+        Returns: A collection.
         """
-        path = Path(path).expanduser().resolve()
-        if not path.exists():
-            raise PathDoesNotExistsError(f"{path} does not exists")
-        if not path.is_dir():
-            raise PathIsNotADirectoryError(f"{path} is not a directory")
-        for mandatory_directory in MANDATORY_DIRECTORIES:
-            mandatory_path = path / mandatory_directory
-            if not mandatory_path.exists() or not mandatory_path.is_dir():
-                raise MissingMandatoryDirectoryError(
-                    f"{path} does not contain the mandatory directory {mandatory_directory}",
-                )
-        return Collection(path)
+        return Collection.from_path(path=Path(path).expanduser().resolve())
```

Comment:
I don't know how it works but for me this is an infinite loop. Have you tested your code ?